### ISSUE TITLE
False positive for `doc_lazy_continuation`

### ISSUE THREAD

**Issue by GuillaumeGomez** [rust-lang/rust-clippy#12941]
### Summary

A valid markdown doc comment is emitting a warning.

In this case, I think it should check if the line is starting with `*` or `-` before emitting the lint.

### Lint Name

doc_lazy_continuation

### Reproducer

I tried this code:

```rust
/// stat
///
/// - 1: bla
/// - 6: Intel PECI
/// Not all types are supported by all chips
pub struct B;
```

I saw this happen:

```
error: doc list item missing indentation
  --> src/unix/linux/component.rs:59:9
   |
59 |     /// Not all types are supported by all chips
```

I expected to see this happen:

Nothing

### Version

```text
rust version 1.81.0-nightly (3cf924b93 2024-06-15)
```

**Comment by Jarcho** [comment]
That's exactly the thing it's supposed to check for. That comment renders the same as:

```rust
/// stat
///
/// - 1: bla
/// - 6: Intel PECI Not all types are supported by all chips
```

**Comment by GuillaumeGomez** [comment]
You are absolutely right. Then I'm glad this new lint exists. :)

Thanks!
